mozillaspeechlibrary/src/main/java/com/mozilla/speechlibrary/networking/GeckoNetworking.java

```diff
@@ -0,0 +1,94 @@
+package com.mozilla.speechlibrary.networking;
+
+import android.content.Context;
+import android.os.Handler;
+import android.os.Looper;
+
+import com.mozilla.speechlibrary.MozillaSpeechService;
+import com.mozilla.speechlibrary.STTResult;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.mozilla.geckoview.GeckoRuntime;
+import org.mozilla.geckoview.GeckoWebExecutor;
+import org.mozilla.geckoview.WebRequest;
+
+import java.io.ByteArrayOutputStream;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+
+public class GeckoNetworking extends Networking {
+
+    private Handler mHandler;
+
+    public GeckoNetworking(Context aContext, MozillaSpeechService aSpeechService) {
+        super(aContext, aSpeechService);
+
+        mHandler = new Handler(Looper.getMainLooper());
+    }
+
+    @Override
+    public void doSTT(final ByteArrayOutputStream baos, final NetworkSettings mNetworkSettings) {
+        if (cancelled) {
+            mSpeechService.notifyListeners(MozillaSpeechService.SpeechState.CANCELED, null);
+            return;
+        }
+
+        mHandler.post(() -> {
+            GeckoWebExecutor executor = new GeckoWebExecutor(GeckoRuntime.getDefault(mContext));
+            ByteBuffer input = ByteBuffer.allocateDirect(baos.size());
+            input.put(baos.toByteArray());
+            WebRequest request = new WebRequest.Builder(STT_ENDPOINT)
+                    .body(input)
+                    .addHeader("Accept-Language-STT", mNetworkSettings.mLanguage)
```

Comment:
After adding `.method("POST")` here, and debugging locally along the server, I started seeing the proper json containing the transcription being returned